.github/workflows/add-lockdown-label.yml

```diff
@@ -0,0 +1,78 @@
+name: Add Branch Lockdown Label to PRs
+
+on:
+  schedule:
+    # Run daily at 00:00 UTC
+    - cron: '0 23 * * *'
+  workflow_dispatch: # Allows manual triggering of the workflow
+
+permissions:
+  actions: write # For managing the operation state cache
+  issues: write
+
+jobs:
+  add-label:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install jq
+        run: sudo apt-get install -y jq
+
+      - name: Add label to PRs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Determine the third Tuesday of the current month
+          third_tuesday=$(date -d "$(date +%Y-%m-01) +14 days" +%Y-%m-%d)
+          while [ $(date -d "$third_tuesday" +%u) -ne 2 ]; do
+            third_tuesday=$(date -d "$third_tuesday + 1 day" +%Y-%m-%d)
+          done
+          
+          # Determine the first Tuesday of the current month
+          first_tuesday=$(date -d "$(date +%Y-%m-01)" +%Y-%m-%d)
+          while [ $(date -d "$first_tuesday" +%u) -ne 2 ]; do
+            first_tuesday=$(date -d "$first_tuesday + 1 day" +%Y-%m-%d)
+          done
+
+          # Get current date
+          current_date=$(date +%Y-%m-%d)
+
+          echo "Current Date: $current_date"
+          echo "Third Tuesday of the month: $third_tuesday"
+          echo "First Tuesday of the month: $first_tuesday"
+
+          # Check if the current date is after the third Tuesday of this month or before the first Tuesday of this month
+          if [[ "$current_date" > "$third_tuesday" || "$current_date" < "$first_tuesday" ]]; then
+            echo "Within the label period. Adding labels to PRs..."
+
+            # Get all open PRs created by app/dotnet-maestro and targeting branches release/8* and release/9* excluding release/9.0.3xx
+            echo "Running gh pr list query..."
+            prs=$(gh pr list --state open --limit 100 --json number,baseRefName,author,labels)
+            echo "Total PRs Count: $(echo "$prs" | jq length)"
+            echo "PRs JSON: $prs"
+
+            echo "Filtering PRs created by app/dotnet-maestro..."
+            prs_by_author=$(echo "$prs" | jq '[.[] | select(.author.login == "app/dotnet-maestro")]')
+            echo "PRs by app/dotnet-maestro JSON: $prs_by_author"
+            echo "Count of PRs by app/dotnet-maestro: $(echo "$prs_by_author" | jq length)"
+
+            echo "Filtering PRs targeting release/8* and release/9* excluding release/9.0.3xx..."
+            prs_targeting_branches=$(echo "$prs_by_author" | jq '[.[] | select(.baseRefName | test("^release/[89].*")) | select(.baseRefName | test("^release/9.0.3xx") | not)]')
```

Comment:
This will change (somewhat) regularly as we create new branches. If we need a branch filter, it should just be a separate file (or variable/parameter) to this action. But, I don't know GitHub Actions to know what makes the most sense here.

Reply:
I'd worry more about a variable if we were using this in other places. We'd need to change it wherever it lives so I don't think it matters if it's in this file or elsewhere.